operator/cmd/configurator/configurator.go

```diff
@@ -128,15 +129,52 @@ func (c *configuratorConfig) String() string {
 var errorMissingEnvironmentVariable = errors.New("missing environment variable")
 
 func Command() *cobra.Command {
-	return &cobra.Command{
+	var (
+		cloudSecretsEnabled          bool
+		cloudSecretsPrefix           string
+		cloudSecretsAWSRegion        string
+		cloudSecretsAWSRoleARN       string
+		cloudSecretsGCPProjectID     string
+		cloudSecretsAzureKeyVaultURI string
+	)
+	cmd := &cobra.Command{
 		Use:     "configurator",
 		Short:   "Configure redpanda.yaml based on Pod metadata",
 		Aliases: []string{"configure"},
-		Run:     run,
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx := cmd.Context()
+			run(
+				ctx,
+				cloudSecretsEnabled,
+				cloudSecretsPrefix,
+				cloudSecretsAWSRegion,
+				cloudSecretsAWSRoleARN,
+				cloudSecretsGCPProjectID,
+				cloudSecretsAzureKeyVaultURI,
+			)
+		},
 	}
+
+	// secret store related flags
+	cmd.Flags().BoolVar(&cloudSecretsEnabled, "enable-cloud-secrets", false, "Set to true if config values can reference secrets from cloud secret store")
+	cmd.Flags().StringVar(&cloudSecretsPrefix, "cloud-secrets-prefix", "", "Prefix for all names of cloud secrets")
+	cmd.Flags().StringVar(&cloudSecretsAWSRegion, "cloud-secrets-aws-region", "", "AWS Region in which the secrets are stored")
+	cmd.Flags().StringVar(&cloudSecretsAWSRoleARN, "cloud-secrets-aws-role-arn", "", "AWS role ARN to assume when fetching secrets")
+	cmd.Flags().StringVar(&cloudSecretsGCPProjectID, "cloud-secrets-gcp-project-id", "", "GCP project ID in which the secrets are stored")
+	cmd.Flags().StringVar(&cloudSecretsAzureKeyVaultURI, "cloud-secrets-azure-key-vault-uri", "", "Azure Key Vault URI in which the secrets are stored")
+
+	return cmd
 }
 
-func run(cmd *cobra.Command, args []string) {
+func run(
+	ctx context.Context,
+	cloudSecretsEnabled bool,
```

Comment:
Same comment here. Given that this is being done in two different places it might be worth while to expose a helper method from the expander package. (Though that requires a separate commit to be merged and then used 🤢, some where under `cmd` feels like an okay middle ground)